### PR TITLE
Adds Incident Location client-side form + backend

### PR DESCRIPTION
**Pull request by el-mapache**
[Link to ZenHub issue.](https://github.com/18F/crt-portal/issues/67)

## What does this change?

* Adds incident location model, view, url entry point, and related migrations!

## Screenshots (for front-end PR):
Screen Shot 2019-11-26 at 9 53 01 AM

### Reviewer

+ [ ] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.)
+ [ ] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

## Notes for reviewer:

See [PR instructions doc](https://github.com/usdoj/crt-portal/blob/master/docs/pull_requests.md) for full pull request review instructions.

### CODE REVIEW

File: crt_portal/static/sass/custom/typography.scss [hunk 1]
```diff
@@ -1,19 +1,19 @@
 // Header Styles
 h1 {
-  @include u-font('heading', '2xl')
+  @include u-font('heading', '2xl');
```
Comment (alexsoble) [inline crt_portal/static/sass/custom/typography.scss]: Nice catches here!

File: crt_portal/cts_forms/templates/forms/report_location.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{% extends "forms/report_base.html" %}
+
+{% block form_questions %}
+  {% include 'forms/grouped_questions.html' %}
```
Comment (alexsoble) [inline crt_portal/cts_forms/templates/forms/report_location.html]: Hey, you were able to use this as a factored-out template! Nice!

File: crt_portal/cts_forms/forms.py [hunk 3]
```diff
@@ -1,11 +1,22 @@
...
+    ModelMultipleChoiceField, Select
 from .question_group import QuestionGroup
-from .widgets import UsaRadioSelect, UsaCheckboxSelectMultiple, CrtRadioArea
+from .widgets import UsaRadioSelect, UsaCheckboxSelectMultiple, CrtRadioArea, CrtDropdown
 from .models import Report, ProtectedClass
-from .model_variables import EMPLOYER_SIZE_CHOICES, PUBLIC_OR_PRIVATE_SCHOOL_CHOICES, RESPONDENT_TYPE_CHOICES, \
-    PUBLIC_OR_PRIVATE_EMPLOYER_CHOICES, PUBLIC_OR_PRIVATE_FACILITY_CHOICES, PUBLIC_OR_PRIVATE_HEALTHCARE_CHOICES, \
-    PROTECTED_CLASS_CHOICES, PROTECTED_CLASS_ERROR, PRIMARY_COMPLAINT_CHOICES, PRIMARY_COMPLAINT_CHOICES_TO_EXAMPLES, PRIMARY_COMPLAINT_CHOICES_TO_HELPTEXT
+from .model_variables import (
+    EMPLOYER_SIZE_CHOICES,
```
Comment (alexsoble) [inline crt_portal/cts_forms/forms.py]: Thanks for the code style improvement here!

File: crt_portal/cts_forms/templates/forms/grouped_questions.html [hunk 4]
```diff
@@ -0,0 +1,46 @@
+{% for question_group in wizard.form.question_groups %}
+  <fieldset class="usa-fieldset margin-bottom-5">
+    <!-- TODO: REMOVE using an h2 here over a legend allows us to benefit from the auto styling of
```
Comment (alexsoble) [inline crt_portal/cts_forms/templates/forms/grouped_questions.html]: Hmmm ... I don't see an `<h2>` or a `<legend`> in this file, so should this code comment move somewhere else? Does `TODO: REMOVE` mean that you want to remove this comment before merging the PR?

**Comment by alexsoble** [comment]
@el-mapache Left some in-line comments above ☝️ before I start digging into the functionality and checking against the ACs.

Also, I'm seeing a merge conflict with `develop`; would you mind patching that up for a clean merge? Thanks!

File: crt_portal/cts_forms/templates/forms/grouped_questions.html (hunk 4 above)
Comment (el-mapache) [reply crt_portal/cts_forms/templates/forms/grouped_questions.html]: Yes, the todo is a note to myself to remove this before the WIP status is removed. There _was_ an `h2`, (now an `h3`), but the comment relates to `h` tags in general 😄

File: crt_portal/cts_forms/views.py [hunk 5]
```diff
@@ -98,6 +102,10 @@ def get_context_data(self, form, **kwargs):
             context.update({
                 'page_note': 'Continued'
             })
+        elif current_step_name == 'Location':
+            context.update({
+                'page_note': 'Providing details on where this occured helps us properly review your issue and get it to the right people within the Civil Rights Division.'
```
Comment (alexsoble) [inline crt_portal/cts_forms/views.py]: Spelling fix needed: `"occured" => "occurred"`.

**Comment by alexsoble** [comment]
@el-mapache I'm noticing that this PR is introducing a regression in how group legends are styled typographically, compare this branch with dev:

### This branch

your-name-this-branch

### Dev

your-name-dev

**Comment by el-mapache** [comment]
@alexsoble Yes, I'm planning on bringing this up with the design team. The classes specified in our design mocks automatically add the serif font when used. Not sure if that is the intended effect or not 😒

**Comment by el-mapache** [comment]
@LindsayYoung Love to have your eyes on this as well! Also, this PR does still have the initial commit from Darren's PR, and I'm wondering if we should merge with that commit present?

**Comment by alexsoble** [comment]
@el-mapache Reviewing against the ACs now.

**Comment by alexsoble** [comment]
I checked against the ACs — overall, looks really good to me! 

I did find one small wording error, which I'll comment on inline below.

Also, in the visual mocks, the location name and street address input fields are longer and stretch all the way to end at the spot where the "State" field ends. I don't mind the shorter input fields but I wonder what the designers' perspective will be.

File: crt_portal/cts_forms/model_variables.py [hunk 6]
```diff
@@ -261,3 +261,9 @@
     ("AE", "Armed Forces Middle East "),
     ("AP", "Armed Forces Pacific "),
 )
+
+WHERE_ERRORS = (
+    ('location_name', 'Please enter the name of this location where this took place.'),
```
Comment (alexsoble) [inline crt_portal/cts_forms/model_variables.py]: This line should be: "Please enter the name of the location where this took place."

Also, mega-small thing, but consistent periods between the three error messages. 

I like the way you factored these into a variable by the way!

Comment (el-mapache) [reply crt_portal/cts_forms/model_variables.py]: No, there should definitely be a period at the end of each, thanks for the catch!

**Comment by el-mapache** [comment]
@alexsoble Would your typically tag the designers to pull this down and check it out, or merge and let them bring up issues once they view on the dev server?

**Comment by alexsoble** [comment]
@el-mapache Merging and letting designers check out the work on dev is a good strategy! Sometimes I set up a quick meeting with one or both of the designers and share my screen to show work-in-progress. 

Neither of the designers has the app running locally, to my knowledge — I met with Jacklynn once about a local setup, but there were hesitatations about installing tools like Docker that require admin permissions.

**Comment by el-mapache** [comment]
@alexsoble Thanks! Just waiting on another PR, then I'll merge and alert design once its reviewable on dev.

**Pull request merged**